src/spikeinterface/preprocessing/filter_gaussian.py

```diff
@@ -7,40 +7,50 @@
 from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
 
 
-class GaussianBandpassFilterRecording(BasePreprocessor):
+class GaussianFilterRecording(BasePreprocessor):
     """
-    Class for performing a bandpass gaussian filtering/smoothing on a recording.
+    Class for performing a gaussian filtering/smoothing on a recording.
+
     This is done by a convolution with a Gaussian kernel, which acts as a lowpass-filter.
-    The highpass-filter can be computed by subtracting the result.
+    A highpass-filter can be computed by subtracting the result of the convolution to
+    the original signal.
+    A bandpass-filter is obtained by substracting the signal smoothed with a narrow
+    gaussian to the signal smoothed with a wider gaussian.
 
-    Here, the bandpass is computed in the Fourier domain to accelerate the computation.
+    Here, convolution is performed in the Fourier domain to accelerate the computation.
 
     Parameters
     ----------
     recording: BaseRecording
         The recording extractor to be filtered.
-    freq_min: float
+    freq_min: float or None
         The lower frequency cutoff for the bandpass filter.
-    freq_max: float
+        If None, the resulting object is a lowpass filter.
+    freq_max: float or None
         The higher frequency cutoff for the bandpass filter.
+        If None, the resulting object is a highpass filter.
     margin_sd: float, default: 5.0
         The number of standard deviation to take for margins.
 
     Returns
     -------
-    gaussian_bandpass_filtered_recording: GaussianBandpassFilterRecording
+    gaussian_filtered_recording: GaussianFilterRecording
         The filtered recording extractor object.
     """
 
-    name = "gaussian_bandpass_filter"
+    name = "gaussian_filter"
 
     def __init__(
         self, recording: BaseRecording, freq_min: float = 300.0, freq_max: float = 5000.0, margin_sd: float = 5.0
     ):
-        sf = recording.sampling_frequency
         BasePreprocessor.__init__(self, recording)
         self.annotate(is_filtered=True)
 
+        if freq_min is None and freq_max is None:
+            raise ValueError("At least one of `freq_min`,`freq_max` should be specified.")
+        if freq_min is not None and freq_max is not None and freq_max <= freq_min:
+            raise ValueError("Expecting `freq_min`<`freq_max`.")
```

Comment:
```suggestion
```
I agree with @DradeAW. Actually specifying `freq_min`>`freq_max` allows to use this as a "band-stop" filter